saleor/graphql/account/mutations/base.py

```diff
@@ -0,0 +1,181 @@
+import graphene
+from django.conf import settings
+from django.contrib.auth.tokens import default_token_generator
+from django.core.exceptions import ValidationError
+from django.utils.encoding import force_bytes
+from django.utils.http import urlsafe_base64_encode
+from graphql_jwt.exceptions import PermissionDenied
+
+from ....account import emails, events as account_events, models
+from ...account.types import Address, AddressInput, User
+from ...core.mutations import (
+    ClearMetaBaseMutation,
+    ModelDeleteMutation,
+    ModelMutation,
+    UpdateMetaBaseMutation,
+)
+
+INVALID_TOKEN = "Invalid or expired token."
+
+
+def send_user_password_reset_email(user, site):
+    context = {
+        "email": user.email,
+        "uid": urlsafe_base64_encode(force_bytes(user.pk)),
+        "token": default_token_generator.make_token(user),
+        "site_name": site.name,
+        "domain": site.domain,
+        "protocol": "https" if settings.ENABLE_SSL else "http",
+    }
+    emails.send_password_reset_email.delay(context, user.email, user.pk)
+
+
+def can_edit_address(user, address):
+    """Determine whether the user can edit the given address.
+
+    This method assumes that an address can be edited by:
+    - users with proper permission (staff)
+    - customers who "own" the given address.
+    """
+    has_perm = user.has_perm("account.manage_users")
+    belongs_to_user = address in user.addresses.all()
+    return has_perm or belongs_to_user
+
+
+class SetPasswordInput(graphene.InputObjectType):
+    token = graphene.String(
+        description=(
+            "A one-time token required to set the password. "
+            "Sent by email using PasswordReset mutation for if staff user "
```

Comment:
Here I would only say "A one-time token required to set the password.". Reset of the description would be moved to meta description.